lib/tower_cli/models/base.py

```diff
@@ -829,11 +830,158 @@ class MonitorableResource(ResourceMethods):
         """A stub method requesting the status of the resource."""
         raise NotImplementedError('This resource does not implement a status '
                                   'method, and must do so.')
 
+    def last_job_data(self, pk=None, **kwargs):
+        """
+        Internal utility function for Unified Job Templates
+        Returns data about the last job run off of that UJT
+        """
+        ujt = self.get(pk, include_debug_header=True, **kwargs)
+
+        # Determine the appropriate inventory source update.
+        if 'current_update' in ujt['related']:
+            debug.log('A current job; retrieving it.', header='details')
+            return client.get(ujt['related']['current_update'][7:]).json()
+        elif ujt['related'].get('last_update', None):
+            debug.log('No current job or update exists; retrieving the most '
+                      'recent.', header='details')
+            return client.get(ujt['related']['last_update'][7:]).json()
+        else:
+            raise exc.NotFound('No related jobs or updates exist.')
+
+    def lookup_stdout(self, pk=None, start_line=None, end_line=None):
+        """
+        Internal utility function to return standard out
+        requires the pk of a unified job
+        """
+        stdout_url = '%s%d/stdout/' % (self.unified_job_type, pk)
+        payload = {
+            'format': 'json', 'content_encoding': 'base64',
+            'content_format': 'ansi'}
+        if start_line:
+            payload['start_line'] = start_line
+        if end_line:
+            payload['end_line'] = end_line
+        debug.log('Requesting a copy of job standard output', header='details')
+        resp = client.get(stdout_url, params=payload).json()
+        content = b64decode(resp['content'])
+
+        return content
+
+    @resources.command
+    @click.option('--start-line', required=False, type=int,
+                  help='Line at which to start printing the standard out.')
+    @click.option('--end-line', required=False, type=int,
+                  help='Line at which to end printing the standard out.')
+    def stdout(self, pk, start_line=None, end_line=None, **kwargs):
+        """
+        Print out the standard out of a unified job to the command line.
+        For Projects, print the standard out of most recent update
+        For Inventory Sources, print standard out of most recent sync
+        For Jobs, print the job's standard out
+        """
+        # resource is Unified Job Template
+        if self.unified_job_type != self.endpoint:
+            unified_job = self.last_job_data(pk, **kwargs)
+            pk = unified_job['id']
+        # resource is Unified Job, but pk not given
+        elif not pk:
+            unified_job = self.get(**kwargs)
+            pk = unified_job['id']
+
+        content = self.lookup_stdout(pk, start_line, end_line)
+        if len(content) > 0:
+            click.echo(content, nl=0)
+
+        return {"changed": False}
```

Comment:
Is it necessary to provide this return value since it doesn't actually contain any info? Because the job stdout might not end with newline and something like this occurs:
```
(tower_cli_devel) dyn-160-39-224-60:tower-cli sitan$ tower-cli job stdout 11
Waiting for results...OK. (changed: false)
```